frontend/public/components/configmaps/ConfigMapForm.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik, FormikHelpers } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { k8sCreate, k8sUpdate } from '@console/internal/module/k8s';
+import { ConfigMapModel } from '@console/internal/models';
+import { history } from '@console/internal/components/utils';
+import { EditorType } from '@console/shared/src/components/synced-editor/editor-toggle';
+import { safeYAMLToJS } from '@console/shared/src/utils/yaml';
+import { getConfigmapData, getConfigMapInitialValues, validationSchema } from './configmap-utils';
+import { ConfigMapFormEditor } from './ConfigMapFormEditor';
+import { ConfigMap, ConfigMapFormInitialValues } from './types';
+
+export interface ConfigMapProps {
+  title: string;
+  namespace: string;
+  name: string;
+  configMap: ConfigMap;
+  isCreateFlow: boolean;
+}
+
+const ConfigmapForm: React.FC<ConfigMapProps> = ({
+  name,
+  namespace,
+  title,
+  configMap,
+  isCreateFlow,
+}) => {
+  const { t } = useTranslation();
+  const [initialValues] = React.useState(
+    getConfigMapInitialValues(namespace, configMap, isCreateFlow),
+  );
+  const handleSubmit = (
+    values: ConfigMapFormInitialValues,
+    actions: FormikHelpers<ConfigMapFormInitialValues>,
+  ) => {
+    let resourceCall;
+    let configMapYaml: ConfigMap;
+    try {
+      // Use YAML also as base when submitting the form
+      configMapYaml = safeYAMLToJS(values.yamlData);
+    } catch (err) {
+      actions.setStatus({
+        submitSuccess: '',
+        submitError: t('public~Invalid YAML - {{err}}', { err }),
+      });
+      return;
+    }
```

Comment:
do we need try / catch block as `safeYAMLToJS` has handled it and will return `{}` otherwise

Reply:
No looks like we dont need try/catch as safeYAMLToJS already handles try/catch in its implementation. Removed it.